ibis/expr/types/strings.py

```diff
@@ -1,18 +1,836 @@
+        import ibis.expr.operations as ops
+
+        return ops.StringLength(self).to_expr()
+
+    def lower(self) -> StringValue:
+        """Convert string to all lowercase.
+
+        Returns
+        -------
+        StringValue
+            Lowercase string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.Lowercase(self).to_expr()
+
+    def upper(self) -> StringValue:
+        """Convert string to all uppercase.
+
+        Returns
+        -------
+        StringValue
+            Uppercase string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.Uppercase(self).to_expr()
+
+    def reverse(self) -> StringValue:
+        """Reverse the characters of a string.
+
+        Returns
+        -------
+        StringValue
+            Reversed string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.Reverse(self).to_expr()
+
+    def ascii_str(self) -> ir.IntegerValue:
+        """Return the numeric ASCII code of the first character of a string.
+
+        Returns
+        -------
+        IntegerValue
+            ASCII code of the first character of the input
+        """
+        import ibis.expr.operations as ops
+
+        return ops.StringAscii(self).to_expr()
+
+    def strip(self) -> StringValue:
+        """Remove whitespace from left and right sides of a string.
+
+        Returns
+        -------
+        StringValue
+            Stripped string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.Strip(self).to_expr()
+
+    def lstrip(self) -> StringValue:
+        """Remove whitespace from the left side of string.
+
+        Returns
+        -------
+        StringValue
+            Left-stripped string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.LStrip(self).to_expr()
+
+    def rstrip(self) -> StringValue:
+        """Remove whitespace from the right side of string.
+
+        Returns
+        -------
+        StringValue
+            Right-stripped string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.RStrip(self).to_expr()
+
+    def capitalize(self) -> StringValue:
+        """Capitalize the input string.
+
+        Returns
+        -------
+        StringValue
+            Capitalized string
+        """
+        import ibis.expr.operations as ops
+
+        return ops.Capitalize(self).to_expr()
+
+    initcap = capitalize
+
+    def __contains__(self, *_: Any) -> bool:
+        raise TypeError("Use string_expr.contains(arg)")
+
+    def contains(self, substr: str | StringValue) -> ir.BooleanValue:
+        """Return whether the expression contains `substr`.
+
+        Parameters
+        ----------
+        substr
+            Substring for which to check
+
+        Returns
+        -------
+        BooleanValue
+            Boolean indicating the presence of `substr` in the expression
+        """
+        return self.find(substr) >= 0
```

Comment:
Eventually if there are backends with specific support for this we may want to introduce a `StringContains` operator and implement this for backends not supporting it.

Reply:
Agreed. It looks like DuckDB has `instr` and `position(needle in haystack)` for this.

Reply:
Can you create a follow-up ticket?